.github/workflows/pull_request.yaml

```diff
@@ -74,19 +78,44 @@ jobs:
     name: Docker Tests
     needs: [lint]
     runs-on: ubuntu-latest
-
+    
     steps:
-      - name: checkout
+      - name: Checkout repository
         uses: actions/checkout@v4
 
-      - name: docker build
-        run: make docker-build
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver: docker
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          build-args: |
+            KONSTRAINT_VER="${{ env.DOCKER_METADATA_OUTPUT_VERSION }}"
+          push: false
```

Comment:
this doesn't support multi-arch unless you only want to build it and not reuse it (because of the docker driver needed in the setup-buildx-action)

Reply:
That's fine for now, I opened https://github.com/plexsystems/konstraint/issues/586 in case anyone wants to follow up.